feat(curriculum): add blog post card lab (#56165)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/lab-blog-post-card/66eaddd04a9e533fba689001.md

---
id: 66eaddd04a9e533fba689001
title: Design a Blog Post Card
challengeType: 14

> dashedName: "lab-blog-post-card"

Reviewer: ```suggestion
dashedName: lab-blog-post-card
"```

demoType: onClick
---

# --description--

In this lab, you'll practice how to style backgrounds and borders.

**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab.

**User Stories:**

1. You should have a `div` with a class of `blog-post-card` to hold all your card elements.

> 2. You should have an image with a valid `alt` text and the class of `post-img`. You can use `cover-photo.jpg` for the `src` attribute of your image.

Reviewer: The image source should be a cdn image.
It should update in the solutions code as well.

> 3. You should have a `div` with a class of `post-content`. Within the `.post-content` `div`:

Reviewer: Seems a bit better this way regarding readability.
Not something necessary though.

```suggestion
3. You should have a `div` with a class of `post-content`. 
4. Within the `.post-content` `div`:
```

    - You should have a `h2` with a class of `post-title`. Assign some text to act as the title of your blog post.
    - You should have a `p` with a class of `post-excerpt`. Assign some text to summarize the blog post (maximum of 20 words).

>     - You should an have an `a` with a class of `read-more`. Assign the text `Read More` to it.

Reviewer: ```suggestion
    - You should have an `a` with a class of `read-more`. Assign the text `Read More` to it.
```

> 4. The `blog-post-card` class should add a white background, rounded corners, shadow for depth, overflow hidden to round the image borders, and control the card width and text alignment.

Reviewer: The `shadow` property is not being tested as there is no hint for it.

> 5. The `post-img` class should ensure that the image fills the card's width and adds a border at the bottom.

Reviewer: The test and hint for this instruction is missing.

6. The `post-content` class should add padding inside the card for spacing.

> 7. The `post-title` and `post-excerpt` elements within `post-content` should be styled for text color and margin.

Reviewer: The test and hint for this instruction is missing.

8. The `read-more` element should be styled for text color, background color, margin, `inline-block` display, `border-radius`, and `padding`. It should be styled like a button.
9. The `read-more` element should have a hover effect that changes the background color.

> 

Reviewer: this note should be added at the end:

```
**Note:** Be sure to link your stylesheet in your HTML and apply your CSS.
```

You should have a `.blog-post-card` `div`.

```js
assert.exists(document.querySelector("div.blog-post-card"));
```

You should have a `.post-img` `img`. Make sure your image has an `alt` text.

```js
const img = document.querySelector('img.post-img');
assert.exists(img, 'The <img> element with class "post-img" exists');
assert.isString(img.alt, 'The alt text of the <img> element is a string');
assert.isNotEmpty(img.alt, 'The alt text of the <img> element is not empty');
```

You should have a `.post-content` `div`.

```js
assert.exists(document.querySelector("div.post-content"));
```

You should have a `.post-title` `h2`.

```js
assert.exists(document.querySelector("h2.post-title"));
```

You should have a `.post-excerpt` `p`.

```js
assert.exists(document.querySelector("p.post-excerpt"));
```

You should have a `.read-more` `a`.

```js
assert.exists(document.querySelector("a.read-more"));
```

You should have a `.read-more` `a` should have a `Read More` text.

```js
const link = document.querySelector('a.read-more');
assert.exists(link, 'The <a> element with class "read-more" exists');
assert.strictEqual(link.textContent, 'Read More', 'The <a> element has the correct text');
```

> `blog-post-card` should have a `border-radius` property with a value (should not be 0 or a negative value).

Reviewer 2: numbers should be in backticks

```suggestion
`blog-post-card` should have a `border-radius` property with a value (should not be `0` or a negative value).
```

```js
const card = document.querySelector('.blog-post-card');
assert.exists(card, 'The element with class "blog-post-card" exists');

const borderRadius = getComputedStyle(card).borderRadius;
assert.notStrictEqual(borderRadius, '0px', 'The card has rounded corners');
```

`blog-post-card` should have a white background.

```js
const card = document.querySelector('.blog-post-card');
assert.exists(card, 'The element with class "blog-post-card" exists');

const backgroundColor = getComputedStyle(card).backgroundColor;
const whiteColors = ['rgb(255, 255, 255)', 'rgba(255, 255, 255, 1)', '#ffffff', '#fff', 'white'];

> assert.include(whiteColors, backgroundColor.toLowerCase(), 'The card has a white background');

Reviewer 3: you can use `oneOf` here, as you are testing `backgroundColor`, it's the thing that should be all on the left
```suggestion
assert.oneOf(backgroundColor.toLowerCase(), whiteColors, 'The card has a white background');
```

```

You should target `.blog-post-card` and set its `width` property.

```js

> assert(new __helpers.CSSHelp(document).getStyle(".blog-post-card")?.getPropertyValue("width"));

Reviewer 3: ```suggestion
assert.exists(new __helpers.CSSHelp(document).getStyle(".blog-post-card")?.getPropertyValue("width"));
```

```

You should target `.blog-post-card` and set its `text-align` property.

```js

> assert(new __helpers.CSSHelp(document).getStyle(".blog-post-card")?.getPropertyValue("text-align"));

Reviewer 3: it's preferred to never use `assert` alone
```suggestion
assert.exists(new __helpers.CSSHelp(document).getStyle(".blog-post-card")?.getPropertyValue("text-align"));
```

```

You should target `.blog-post-card` and set its `overflow` property.

```js

> assert(new __helpers.CSSHelp(document).getStyle(".blog-post-card")?.getPropertyValue("overflow"));

Reviewer 3: ```suggestion
assert.exists(new __helpers.CSSHelp(document).getStyle(".blog-post-card")?.getPropertyValue("overflow"));
```

> You should target `.blog-post-card` and set its `overflow` property.
> 
> ```js
> assert.exists(new __helpers.CSSHelp(document).getStyle(".blog-post-card")?.getPropertyValue("overflow"));
> ```

Reviewer: ````suggestion
You should target `.blog-post-card` and set its `overflow` property to `hidden`.

```js
const overflowVal = new __helpers.CSSHelp(document).getStyle(".blog-post-card")?.getPropertyValue("overflow");
assert.equal(overflowVal, "hidden");
```
````

  return boxShadow && boxShadow !== 'none';
}

assert.isTrue(hasBoxShadow(card), 'The .blog-post-card should have a box-shadow property set.');
```

You should target `.blog-post-card` and set its `overflow` property to `hidden`.

```js
const overflowVal = new __helpers.CSSHelp(document).getStyle(".blog-post-card")?.getPropertyValue("overflow");
assert.equal(overflowVal, "hidden");
```

You should target `.post-content` and set its `padding` property.

```js
assert.exists(new __helpers.CSSHelp(document).getStyle(".post-content")?.getPropertyValue("padding"));
```

`.read-more` should have a hover effect.

```js
const link = document.querySelector('.read-more');
assert.exists(link, 'The element with class "read-more" exists');

const stylesheet = [...document.styleSheets].find(sheet => [...sheet.rules].some(rule => rule.selectorText === '.read-more:hover'));
assert.exists(stylesheet, 'The hover rule for .read-more exists');

const hoverRule = [...stylesheet.rules].find(rule => rule.selectorText === '.read-more:hover');
const hoverBackgroundColor = hoverRule.style.backgroundColor;
assert.isNotEmpty(hoverBackgroundColor, 'The hover effect changes the background color');
```

> `post-img` should fill the card's `width` and have a `border-bottom` value.

Author: Can you check again? I used the code above and the test passed.

![image](https://github.com/user-attachments/assets/977e7db5-7a6b-47eb-a781-bca91a7858fe)
![image](https://github.com/user-attachments/assets/ba62ab45-c060-47c8-b037-fc44c216fa02)

assert.exists(stylesheet);

const hoverRule = [...stylesheet.rules].find(rule => rule.selectorText === '.read-more:hover');
const hoverBackgroundColor = hoverRule.style.backgroundColor;
assert.isNotEmpty(hoverBackgroundColor);
```

`.read-more` should be styled like a button. 

```js
const readMore = document.querySelector('.read-more');
assert.exists(readMore);

const readMoreDisplay = getComputedStyle(readMore).display;
assert.strictEqual(readMoreDisplay, 'inline-block');

const readMoreMargin = getComputedStyle(readMore).marginTop;
assert.notStrictEqual(readMoreMargin, '0px');

const readMorePadding = getComputedStyle(readMore).padding;
assert.notStrictEqual(readMorePadding, '0px');

const readMoreBackgroundColor = getComputedStyle(readMore).backgroundColor;
assert.notStrictEqual(readMoreBackgroundColor, 'rgba(0, 0, 0, 0)');

const readMoreTextColor = getComputedStyle(readMore).color;
assert.notStrictEqual(readMoreTextColor, 'rgba(0, 0, 0, 0)');

const readMoreBorderRadius = getComputedStyle(readMore).borderRadius;
assert.notStrictEqual(readMoreBorderRadius, '0px');
```

> `post-img` should fill the card's `width` and have a `border-bottom` value.

Reviewer 3: ```suggestion
`.post-img` should fill the card's `width` and have a `border-bottom` value.
```

it's the class name

```js
const postImg = document.querySelector('.post-img');
assert.exists(postImg);

const imgWidth = getComputedStyle(postImg).width;
const cardWidth = getComputedStyle(postImg.parentElement).width;
assert.strictEqual(imgWidth, cardWidth);

const imgBorderBottom = getComputedStyle(postImg).borderBottomWidth;
assert.notStrictEqual(imgBorderBottom, '0px');
```

> Ensure `post-title` and `post-excerpt` have text color and margin.

Reviewer 3: ```suggestion
Ensure `.post-title` and `.post-excerpt` have text color and margin.
```

assert.isNotEmpty(postTitle.textContent);
```

You should have a `.post-excerpt` `p`.

```js
assert.exists(document.querySelector("p.post-excerpt"));
```

You should have a `.read-more` `a`.

```js
assert.exists(document.querySelector("a.read-more"));
```

You should have a `.read-more` `a` should have a `Read More` text.

```js
const link = document.querySelector('a.read-more');
assert.exists(link);
assert.strictEqual(link.textContent, 'Read More');
```

`blog-post-card` should have a `border-radius` property with a value (should not be `0` or a negative value).

```js
const card = document.querySelector('.blog-post-card');
assert.exists(card);

const borderRadius = getComputedStyle(card).borderRadius;
assert.notStrictEqual(borderRadius, '0px');
```

`blog-post-card` should have a white background.

```js
const card = document.querySelector('.blog-post-card');
assert.exists(card);

const backgroundColor = getComputedStyle(card).backgroundColor;
const whiteColors = ['rgb(255, 255, 255)', 'rgba(255, 255, 255, 1)', '#ffffff', '#fff', 'white'];

assert.oneOf(backgroundColor.toLowerCase(), whiteColors);
```

You should target `.blog-post-card` and set its `width` property.

```js

> assert.exists(new __helpers.CSSHelp(document).getStyle(".blog-post-card")?.getPropertyValue("width"));

Reviewer 3: ```suggestion
assert.isNotEmpty(new __helpers.CSSHelp(document).getStyle(".blog-post-card")?.getPropertyValue("width"));
```

```

You should target `.blog-post-card` and set its `text-align` property.

```js

> assert.exists(new __helpers.CSSHelp(document).getStyle(".blog-post-card")?.getPropertyValue("text-align"));

Reviewer 3: ```suggestion
assert.isNotEmpty(new __helpers.CSSHelp(document).getStyle(".blog-post-card")?.getPropertyValue("text-align"));
```

`exists` does not give the desired result

```

You should target `.blog-post-card` and set its shadow property.

```js
const card = document.querySelector('.blog-post-card');

function hasBoxShadow(element) {
  const boxShadow = getComputedStyle(element).getPropertyValue('box-shadow');
  return boxShadow && boxShadow !== 'none';
}

assert.isTrue(hasBoxShadow(card));
```

You should target `.blog-post-card` and set its `overflow` property to `hidden`.

```js
const overflowVal = new __helpers.CSSHelp(document).getStyle(".blog-post-card")?.getPropertyValue("overflow");
assert.equal(overflowVal, "hidden");
```

You should target `.post-content` and set its `padding` property.

```js

> assert.exists(new __helpers.CSSHelp(document).getStyle(".post-content")?.getPropertyValue("padding"));

Reviewer 3: ```suggestion
assert.isNotEmpty(new __helpers.CSSHelp(document).getStyle(".post-content")?.getPropertyValue("padding"));
```

assert.notStrictEqual(readMoreTextColor, 'rgba(0, 0, 0, 0)');

const readMoreBorderRadius = getComputedStyle(readMore).borderRadius;
assert.notStrictEqual(readMoreBorderRadius, '0px');
```

`post-img` should fill the card's `width` and have a `border-bottom` value.

```js
const postImg = document.querySelector('.post-img');
assert.exists(postImg);

const imgWidth = getComputedStyle(postImg).width;
const cardWidth = getComputedStyle(postImg.parentElement).width;
assert.strictEqual(imgWidth, cardWidth);

const imgBorderBottom = getComputedStyle(postImg).borderBottomWidth;
assert.notStrictEqual(imgBorderBottom, '0px');
```

Ensure `post-title` and `post-excerpt` have text color and margin.

```js
const title = document.querySelector('.post-title');
const excerpt = document.querySelector('.post-excerpt');

function isColorApplied(element) {
  const color = getComputedStyle(element).getPropertyValue('color');
  return color && color !== 'rgba(0, 0, 0, 0)' && color !== 'transparent'; 
}

function isMarginApplied(element) {
  const margin = getComputedStyle(element).getPropertyValue('margin');
  return margin && margin !== '0px'; 
}

const titleStyles = getComputedStyle(title);

> const excerptStyles = getComputedStyle(excerpt);

Reviewer 3: I don't find where this two are used

---
id: 66eaddd04a9e533fba689001
title: Design a Blog Post Card
challengeType: 14

> dashedName: lab-blog-post-card

Reviewer: ```suggestion
dashedName:  design-a-blog-post-card
```

Reviewer 2: As mentioned in chat, this will be updated to the mongo id.

demoType: onClick
---

# --description--

In this lab, you'll practice how to style backgrounds and borders by creating a blog post card.

**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab.

**User Stories:**

1. You should have a `div` with a class of `blog-post-card` to hold all your card elements.

> 2. You should have an image with a valid `alt` text and the class of `post-img`. You can use `https://cdn.freecodecamp.org/curriculum/labs/cover-photo.jpg` for the `src` attribute of your image.

Reviewer: It should be mentioned in the user stories that `post-img` should be inside the `blog-post-card` div. The test for it already exists, so updating the instruction would be better.

> 3. You should have a `div` with a class of `post-content` within your `blog-post-card` `div`.  
>    Within the `.post-content` `div`:

Reviewer: ```suggestion
3. You should have a `div` with a class of `post-content` within your `blog-post-card` `div`. Within the `.post-content` `div`:
```

   - You should have an `h2` with a class of `post-title`. Assign some text to act as the title of your blog post.

>    - You should have a `p` with a class of `post-excerpt`. Assign some text to summarize the blog post (maximum of 20 words).

Reviewer: Max of 20 words isn't being tested anywhere.I'd vote to remove this limit.

```js
const card = document.querySelector("div.blog-post-card");
assert.exists(card);
const content = card.querySelector("div.post-content");
assert.exists(content);
```

You should have a `.post-title` `h2`. Make sure it has some text content.

```js
const card = document.querySelector("div.blog-post-card");
assert.exists(card);

const content = card.querySelector("div.post-content");
assert.exists(content);

const postTitle = content.querySelector("h2.post-title");
assert.exists(postTitle);
assert.isString(postTitle.textContent);
assert.isNotEmpty(postTitle.textContent);
```

You should have a `.post-excerpt` `p`.

```js
const card = document.querySelector("div.blog-post-card");
assert.exists(card);

const content = card.querySelector("div.post-content");
assert.exists(content);

const postExcerpt = content.querySelector("p.post-excerpt");
assert.exists(postExcerpt);
```

You should have a `.read-more` `a`.

```js
const card = document.querySelector("div.blog-post-card");
assert.exists(card);

const content = card.querySelector("div.post-content");
assert.exists(content);

const readMore = content.querySelector("a.read-more");
assert.exists(readMore);
```

> You should have a `.read-more` `a` should have a `Read More` text.

Reviewer: ```suggestion
Your `.read-more` `a` should have a `Read More` text.
```

assert.exists(stylesheet);

const hoverRule = [...stylesheet.rules].find(rule => rule.selectorText === '.read-more:hover');
const hoverBackgroundColor = hoverRule.style.backgroundColor;
assert.isNotEmpty(hoverBackgroundColor);
```

`.read-more` should be styled like a button. 

```js
const readMore = document.querySelector('.read-more');
assert.exists(readMore);

const readMoreDisplay = getComputedStyle(readMore).display;
assert.strictEqual(readMoreDisplay, 'inline-block');

const readMoreMargin = getComputedStyle(readMore).marginTop;
assert.notStrictEqual(readMoreMargin, '0px');

const readMorePadding = getComputedStyle(readMore).padding;
assert.notStrictEqual(readMorePadding, '0px');

const readMoreBackgroundColor = getComputedStyle(readMore).backgroundColor;
assert.notStrictEqual(readMoreBackgroundColor, 'rgba(0, 0, 0, 0)');

const readMoreTextColor = getComputedStyle(readMore).color;
assert.notStrictEqual(readMoreTextColor, 'rgba(0, 0, 0, 0)');

const readMoreBorderRadius = getComputedStyle(readMore).borderRadius;
assert.notStrictEqual(readMoreBorderRadius, '0px');

> ```

Reviewer: It's better to write these tests separately to match the individual style in these user stories:

8. The `read-more` element should be styled like a button and have:

   - A text color other than the default.
   - Background color.
   - Margin on all sides.
   - A `display` property of `inline-block`.
   - Rounded corners.
   - Padding on all sides.

title: Design a Blog Post Card
challengeType: 14
dashedName: lab-blog-post-card
demoType: onClick
---

# --description--

In this lab, you'll practice how to style backgrounds and borders by creating a blog post card.

**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab.

**User Stories:**

1. You should have a `div` with a class of `blog-post-card` to hold all your card elements.
2. You should have an image with a valid `alt` text and the class of `post-img`. You can use `https://cdn.freecodecamp.org/curriculum/labs/cover-photo.jpg` for the `src` attribute of your image.
3. You should have a `div` with a class of `post-content`.  
   Within the `.post-content` `div`:

   - You should have a `h2` with a class of `post-title`. Assign some text to act as the title of your blog post.
   - You should have a `p` with a class of `post-excerpt`. Assign some text to summarize the blog post (maximum of 20 words).
   - You should have an `a` with a class of `read-more`. Assign the text `Read More` to it.

4. You should apply the following styles to the `blog-post-card` div:

   - A white colored background.
   - Rounded corners.

>    - Shadow for depth.

Reviewer 4: I don't think `box-shadow` is taught prior to this. If it's not, I think we should remove this part of the user story and the test.

>    - Appropriate `overflow` property to prevent content from overflowing.

Reviewer 4: Likewise, I don't think `overflow` is taught prior to this. We remove this and the test if not.

assert.exists(readMore);

const readMoreDisplay = getComputedStyle(readMore).display;
assert.strictEqual(readMoreDisplay, 'inline-block');

const readMoreMargin = getComputedStyle(readMore).marginTop;
assert.notStrictEqual(readMoreMargin, '0px');

const readMorePadding = getComputedStyle(readMore).padding;
assert.notStrictEqual(readMorePadding, '0px');

const readMoreBackgroundColor = getComputedStyle(readMore).backgroundColor;
assert.notStrictEqual(readMoreBackgroundColor, 'rgba(0, 0, 0, 0)');

const readMoreTextColor = getComputedStyle(readMore).color;
assert.notStrictEqual(readMoreTextColor, 'rgba(0, 0, 0, 0)');

const readMoreBorderRadius = getComputedStyle(readMore).borderRadius;
assert.notStrictEqual(readMoreBorderRadius, '0px');
```

`.post-img` should fill the card's `width` and have a `border-bottom` value.

```js
const postImg = document.querySelector('.post-img');
assert.exists(postImg);

const imgWidth = getComputedStyle(postImg).width;
const cardWidth = getComputedStyle(postImg.parentElement).width;
assert.strictEqual(imgWidth, cardWidth);

const imgBorderBottom = getComputedStyle(postImg).borderBottomWidth;
assert.notStrictEqual(imgBorderBottom, '0px');
```

> Ensure `.post-title` and `post-excerpt` have text color and margin.

Reviewer 4: This test failed for me when I tried to use a selector like this: `.post-title, .post-excerpt`. We should allow that type of selector. Or potentially, you could test the DOM for the output to see if the elements have what you expect. Then you don't need to worry about the selectors they use. Or you could just change the user story to specify to use a `.post-title` selector, and a `.post-excerpt` selector.

Author: Thanks for pointing that out. I've adjusted the test to account for both grouped or individual selectors.

---
id: 66eaddd04a9e533fba689001
title: Design a Blog Post Card
challengeType: 14
dashedName: lab-design-a-blog-post-card
demoType: onClick
---

# --description--

In this lab, you'll practice how to style backgrounds and borders by creating a blog post card.

**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab.

**User Stories:**

1. You should have a `div` with a class of `blog-post-card` to hold all your card elements.

> 2. Within the `blog-post-card` `div`, you should have an image with a valid `alt` text and the class of `post-img`. You can use `https://cdn.freecodecamp.org/curriculum/labs/cover-photo.jpg` for the `src` attribute of your image.

Reviewer 3: ```suggestion
2. Within the `div.blog-post-card`, you should have an image with a valid `alt` text and the class of `post-img`. You can use `https://cdn.freecodecamp.org/curriculum/labs/cover-photo.jpg` for the `src` attribute of your image.
```

> 3. You should have a `div` with a class of `post-content` within your `blog-post-card` `div`.  

Reviewer 3: ```suggestion
3. You should have a `div` with a class of `post-content` within your `div.blog-post-card`.
4. ```

>    Within the `post-content` `div`:

Reviewer 3: ```suggestion
   Within the `.post-content` element:
```

   - You should have an `h2` with a class of `post-title`. Assign some text to act as the title of your blog post.
   - You should have a `p` with a class of `post-excerpt`. Assign some text to summarize the blog post.
   - You should have an `a` with a class of `read-more`. Assign the text `Read More` to it.

> 4. You should apply the following styles to the `blog-post-card` div:

Reviewer 3: ```suggestion
4. You should apply the following styles to the `.blog-post-card` element:
```

   - A white background.
   - Rounded corners.
   - Width of your choice.
   - Text alignment of your choice.

5. The `post-img` class should ensure that the image fills the card's width and adds a border at the bottom.
6. The `post-content` class should add padding inside the card for spacing.

> 7. The `post-title` and `post-excerpt` elements within `post-content` should have a text color other than the default and a margin on all sides.

Reviewer 3: ```suggestion
7. The `.post-title` and `.post-excerpt` elements within `.post-content` should have a text color other than the default and a margin on all sides.
```

> 8. The `read-more` element should be styled like a button and have:

Reviewer 3: ```suggestion
8. The `.read-more` element should be styled like a button and have:
```

   - A text color other than the default.
   - Background color.
   - Margin on all sides.
   - A `display` property of `inline-block`.
   - Rounded corners.
   - Padding on all sides.

> 9. The `read-more` element should have a hover effect that changes the background color.

Reviewer 3: ```suggestion
9. The `.read-more` element should have a hover effect that changes the background color.
```

**Note:** Be sure to link your stylesheet in your HTML and apply your CSS.

# --hints--

> You should have a `.blog-post-card` `div`.

Reviewer 3: ```suggestion
You should have a `div` with a `class` of `blog-post-card`.
```

```js
assert.exists(document.querySelector("div.blog-post-card"));
```

> You should have a `.post-img` `img`. Make sure your image has an `alt` text and `src` value.

Reviewer 3: ```suggestion
You should have a an `img` element with a `class` of `post-img`. Make sure your image has an `alt` text and `src` value.
```

```js
const card = document.querySelector("div.blog-post-card");
assert.exists(card);
const content = card.querySelector("div.post-content");
assert.exists(content);
```

You should have a `.post-title` `h2`. Make sure it has some text content.

```js
const card = document.querySelector("div.blog-post-card");
assert.exists(card);

const content = card.querySelector("div.post-content");
assert.exists(content);

const postTitle = content.querySelector("h2.post-title");
assert.exists(postTitle);
assert.isString(postTitle.textContent);
assert.isNotEmpty(postTitle.textContent);
```

You should have a `.post-excerpt` `p`.

```js
const card = document.querySelector("div.blog-post-card");
assert.exists(card);

const content = card.querySelector("div.post-content");
assert.exists(content);

const postExcerpt = content.querySelector("p.post-excerpt");
assert.exists(postExcerpt);
```

You should have a `.read-more` `a`.

```js
const card = document.querySelector("div.blog-post-card");
assert.exists(card);

const content = card.querySelector("div.post-content");
assert.exists(content);

const readMore = content.querySelector("a.read-more");
assert.exists(readMore);
```

> Your `.read-more` `a` should have a `Read More` text.

Reviewer 3: ```suggestion
Your `a.read-more` should have a `Read More` text.
```

```js

> const card = document.querySelector("div.blog-post-card");
> assert.exists(card);
> 
> const content = card.querySelector("div.post-content");
> assert.exists(content);
> 
> const link = content.querySelector("a.read-more");
> assert.exists(link);
> assert.strictEqual(link.textContent, "Read More");

Reviewer: According to instructions, the order of elements isn't strict.
The updated test would allow the campers to place the `a` tag in any order within the `post-content ` `div`.

```suggestion
const el = document.querySelector("div.post-content a.read-more");
assert.exists(el);
assert.strictEqual(el.textContent, "Read More");
```
